feature-libs/product-configurator/rulebased/components/attribute/price-change/configurator-attribute-price-change.service.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { ObjectComparisonUtils } from '@spartacus/core';
+import { ConfiguratorRouterExtractorService } from '@spartacus/product-configurator/common';
+import { EMPTY, filter, Observable, of, switchMap, tap } from 'rxjs';
+import { ConfiguratorCommonsService } from '../../../core/facade/configurator-commons.service';
+import { Configurator } from '../../../core/model/configurator.model';
+
+@Injectable()
+export class ConfiguratorAttributePriceChangeService {
```

Comment:
This service is not obvious one. Let's add a JSDoc explaining:
- why it's needed and what it's respobsible for 
- mention that it has to be provided in each @Component instances separately in `@Component({ providers[] })` section